File: AtlasWeb/Core/Modules/Tibbi_Surec/Hasta_Gecmisi/SpacialityModules/OldEyeExaminationFormViewModel.cs
//$2FE9A379
using System;
using System.Linq;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class EyeExaminationServiceController
    {
        partial void PreScript_OldEyeExaminationForm(OldEyeExaminationFormViewModel viewModel, EyeExamination eyeExamination, TTObjectContext objectContext)
        {
            viewModel.NoGlassVisSharpLeftVal = viewModel._EyeExamination.NoGlassVisSharpLeftVal != null ? viewModel._EyeExamination.NoGlassVisSharpLeftVal.SharpnessValue : "";
            viewModel.NoGlassVisSharpRightVal = viewModel._EyeExamination.NoGlassVisSharpRightVal != null ? viewModel._EyeExamination.NoGlassVisSharpRightVal.SharpnessValue : "";
            viewModel.GlassVisSharpLeftVal = viewModel._EyeExamination.GlassVisSharpLeftVal != null ? viewModel._EyeExamination.GlassVisSharpLeftVal.SharpnessValue : "";
            viewModel.GlassVisSharpRightVal = viewModel._EyeExamination.GlassVisSharpRightVal != null ? viewModel._EyeExamination.GlassVisSharpRightVal.SharpnessValue : "";
        }
    }
}

namespace Core.Models
{
    public partial class OldEyeExaminationFormViewModel
    {
        public string NoGlassVisSharpLeftVal { get; set; }
        public string NoGlassVisSharpRightVal { get; set; }
        public string GlassVisSharpLeftVal { get; set; }
        public string GlassVisSharpRightVal { get; set; }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Hasta_Gecmisi/SpacialityModules/OldEyeExaminationFormViewModel.ts
//$D8529456
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { EyeExamination } from 'NebulaClient/Model/AtlasClientModel';

import { Type } from 'NebulaClient/ClassTransformer';

export class OldEyeExaminationFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.OldEyeExaminationFormViewModel, Core';

    @Type(() => EyeExamination)
    public _EyeExamination: EyeExamination = new EyeExamination();

    public NoGlassVisSharpLeftVal: string;
    public NoGlassVisSharpRightVal: string;
    public GlassVisSharpLeftVal: string;
    public GlassVisSharpRightVal: string;
}
